src/tainting/Dataflow_tainting.ml

```diff
@@ -962,43 +968,71 @@ let rec check_tainted_lval env (lval : IL.lval) : Taints.t * Lval_env.t =
   report_findings { env with lval_env } findings;
   (taints, lval_env)
 
-and propagate_taint_via_unresolved_java_getters_and_setters env e args
+and propagate_taint_via_java_getters_and_setters_without_definition env e args
     all_args_taints =
   match e with
   | {
    e =
      Fetch
        ({
-          base = Var _obj;
+          base = Var obj;
           rev_offset =
-            [ { o = Dot { IL.ident = method_str, method_tok; id_info; _ }; _ } ];
+            [ { o = Dot { IL.ident = method_str, method_tok; sid; _ }; _ } ];
         } as lval);
    _;
   }
-    when env.lang =*= Lang.Java
-         && Option.is_none !(id_info.id_resolved)
-         && String.length method_str > 3 -> (
+    when env.lang =*= Lang.Java && String.length method_str > 3 -> (
       (* e.g. getFooBar/setFooBar -> fooBar *)
       let prop_str =
         String.uncapitalize_ascii (Str.string_after method_str 3)
       in
-      (* TODO: Fix this with naming info coming from Pro via some hook. *)
-      let prop_name =
+      let mk_prop_name () =
+        (* FIXME: Pro should be autogenerating definitions for these getters/setters,
+         * but that seems to hurt performance and it's still unclear why, so instead
+         * we give taint access to Pro typing info via 'hook_find_attribute_in_class'
+         * and look for the property corresponding to the getter/setter.
+         *
+         * On very large files, allocating a new name every time _could_ have a perf
+         * impact, so we cache them. *)
+        match Hashtbl.find_opt env.java_props (prop_str, sid) with
+        | Some prop_name -> prop_name
+        | None -> (
+            let mk_default_prop_name () =
+              let prop_name =
+                {
+                  ident = (prop_str, method_tok);
+                  sid = G.SId.unsafe_default;
+                  id_info = G.empty_id_info ();
+                }
+              in
+              Hashtbl.add env.java_props (prop_str, sid) prop_name;
+              prop_name
+            in
+            match (!(obj.id_info.id_type), !hook_find_attribute_in_class) with
+            | None, _
+            | _, None ->
+                mk_default_prop_name ()
+            | Some obj_ty, Some hook -> (
+                match hook obj_ty prop_str with
```

Comment:
A bit confused---why do we only care about the type? Not obvious from the function name